Does Not Build on MacOS ARM

the following error is provided when attempting to build on an arm mac, even when cross-compiling to x64 

```
error[E0570]: `"aapcs"` is not a supported ABI for the current target
  --> kernel/src/syscalls/mod.rs:47:5
   |
47 |     pub unsafe fn invoke(&self, i: &Input, o: &mut Output) -> i6...
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error[E0570]: `"aapcs"` is not a supported ABI for the current target
  --> kernel/src/syscalls/mod.rs:88:5
   |
88 |     pub unsafe fn int44(&self, offset: usize, module: &VPathBuf) -> ...
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  ```
  Seems to be an issue with the ```cpu_abi``` proc macro.

--- a/src/kernel-macros/src/cpu_abi.rs
+++ b/src/kernel-macros/src/cpu_abi.rs
@@ -21,7 +21,7 @@ pub fn transform(item: ItemFn) -> syn::Result<TokenStream> {
     #[cfg(target_arch = "x86_64")]
     let abi = "sysv64";
     #[cfg(target_arch = "aarch64")]
-    let abi = "aapcs";
+    let abi = "C";
     let attrs = item.attrs;
     let vis = item.vis;
     let safety = item.sig.unsafety;

--- a/src/kernel/src/main.rs
+++ b/src/kernel/src/main.rs
@@ -216,7 +216,10 @@ fn main() -> ExitCode {
         }
         #[cfg(not(target_arch = "x86_64"))]
         ExecutionEngine::Native => {
-            error!("Native execution engine cannot be used on your machine.");
+            error!(
+                logger,
+                "Native execution engine cannot be used on your machine."
+            );
             return ExitCode::FAILURE;
         }
         ExecutionEngine::Llvm => {

